src/upload/src/Upload.tsx

```diff
@@ -312,21 +314,44 @@ export default defineComponent({
     }
     function handleFileAddition (files: FileList | null, e?: Event): void {
       if (!files) return
-      Array.from(files).forEach((file) => {
-        const fileInfo: FileInfo = {
+      const { onBeforeUpload } = props
+      const fileInfoList: FileInfo[] = Array.from(files).map((file) => {
+        return {
           id: createId(),
           name: file.name,
           status: 'pending',
           percentage: 0,
           file: file,
           url: null
         }
-        doChange(fileInfo, e, {
-          append: true
-        })
       })
-      if (props.defaultUpload) {
-        submit()
+      if (onBeforeUpload) {
+        void Promise.all(
+          fileInfoList.map(async (fileInfo) => {
```

Comment:
```suggestion
          fileInfoList.map((fileInfo) => {
```

这个 async 也应该是没用的，如果是 eslint 报错就 disabled 掉

Reply:
明白了，上面好多应该都是 eslint 编译出来的，我尝试关掉。